dev/com.ibm.ws.jbatch.cdi_fat/util/src/fat/util/JobWaiter.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package fat.util;
+
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import javax.batch.operations.JobOperator;
+import javax.batch.runtime.BatchRuntime;
+import javax.batch.runtime.BatchStatus;
+import javax.batch.runtime.JobExecution;
+
+/**
+ * Utility to start a job with a static reference to a
+ * {@link javax.batch.operations.JobOperator}
+ * and wait for finished state like {@code BatchStatus.COMPLETED} or
+ * {@code BatchStatus.FAILED}.
+ *
+ * The full set of finished statuses is
+ * <li>{@code BatchStatus.COMPLETED}
+ * <li>{@code BatchStatus.FAILED}.
+ * <li>{@code BatchStatus.STOPPED}.
+ * <li>{@code BatchStatus.ABANDONED}.
+ *
+ * It also waits for job exitStatus to be non-null
+ */
+public class JobWaiter {
+
+    private final static Logger logger = Logger.getLogger("test");
+
+    // Full list:
+    //public enum BatchStatus {STARTING, STARTED, STOPPING, STOPPED, FAILED, COMPLETED, ABANDONED }
+    private static Set<BatchStatus> finishedStatuses = new HashSet<BatchStatus>();
+    static {
+        finishedStatuses.add(BatchStatus.STOPPED);
+        finishedStatuses.add(BatchStatus.FAILED);
+        finishedStatuses.add(BatchStatus.COMPLETED);
+        finishedStatuses.add(BatchStatus.ABANDONED);
+    }
+
+    protected static JobOperator jobOp = BatchRuntime.getJobOperator();
+
+    private final int POLL_INTERVAL = 100; // .1 second
+
+    private long timeout;
+
+    /**
+     * @return Time (milliseconds) to wait for job to finish
+     */
+    public long getTimeout() {
+        return timeout;
+    }
+
+    /**
+     * @param timeout Time (milliseconds) to wait for job to finish
+     */
+    public void setTimeout(long timeout) {
+        this.timeout = timeout;
+    }
+
+    /**
+     * @param timeout Time (milliseconds) to wait for job to finish
+     */
+    public JobWaiter() {
+        this(DFLT_TIMEOUT);
+    }
+
+    /**
+     * @param timeout Time (milliseconds) to wait for job to finish
+     */
+    public JobWaiter(long timeout) {
+        super();
+        this.timeout = timeout;
+    }
+
+    private static long DFLT_TIMEOUT = 10000;
+
+    /**
+     * Wait for {@code JobWaiter#timeout} seconds for BOTH of:
+     *
+     * 1) BatchStatus to be one of: STOPPED ,FAILED , COMPLETED, ABANDONED
+     * AND
+     * 2) exitStatus to be non-null
+     *
+     * Returns JobExecution if it ends in <b>COMPLETED</b> status, otherwise throws
+     * <b>IllegalStateException</b>
+     *
+     * @param jobXMLName
+     * @param jobParameters
+     * @return JobExecution (for successfully completing job)
+     * @throws IllegalStateException
+     */
+    public JobExecution completeNewJob(String jobXMLName, Properties jobParameters) throws IllegalStateException {
```

Comment:
Nice to have this in a convenient all-in-one method. Should be easy to extend it to look for alternate completion states, e.g. for a job we want to fail and be restartable.

Reply:
thanks for noticing 👍